proxy/src/compute.rs

```diff
@@ -66,7 +66,7 @@ impl NodeInfo {
         let mut connection_error = None;
         let ports = self.config.get_ports();
         for (i, host) in self.config.get_hosts().iter().enumerate() {
-            let port = ports.get(i).or_else(|| ports.get(0)).unwrap_or(&5432);
+            let port = ports.get(i).or_else(|| ports.first()).unwrap_or(&5432);
```

Comment:
So this picks:
1. the 'i'th element in the ports array
2. the first element in the ports array
3. default 5432

in that order.

I had to look at the tokio postgres docs to convince myself that that's correct. The reason that's correct is that the ports array is supposed to have 0 entries, 1 entry, or as many entries as in the hosts array. See https://docs.rs/tokio-postgres/latest/tokio_postgres/config/struct.Config.html#method.port. Might be worth adding a comment, or a sanity check like [here](https://docs.rs/tokio-postgres/latest/src/tokio_postgres/connect.rs.html#22).

Reply:
Had added both a check and the comment.